src/extension.rs

```diff
@@ -175,10 +175,22 @@ impl ExtensionSequence {
             while offset < len {
                 let esize = source.read_i32()?;
                 let ecode = source.read_i32()?;
-                let data_size = esize as usize - 8;
-                let mut edata = vec![0u8; data_size];
-                source.read_exact(&mut edata)?;
-                extensions.push(Extension::new(esize, ecode, edata));
+
+                let data_size = (esize as usize).saturating_sub(8);
+                // rather than pre-allocating for the data size, this will
+                // pre-allocate up to a more reliable amount and feed the
+                // vector sequentially, to prevent some trivial OOM attacks
```

Comment:
Can you please explain me why this is important to handle? Does it crash when we load an "Inconsistent or unsupported volume size"? If so, is our `panic` better than an OOM? If not, does it simply hang and the OS tries to find memory?

Reply:
[This SO question](https://stackoverflow.com/q/44923306/1233251) might explain it better than me. Currently, the global allocator may simply choose to abort the whole program on OOM. Panics are somewhat recoverable through unwinding, and server applications are usually prepared for catching panics, but an abort will translate to a KILL signal or equivalent, and it's completely irrecoverable. What I fear here is that we'll have server applications receiving NIfTI files from client applications and I wouldn't want them to be able to crash the server so easily, by just tweaking the `dim` field.

This code change mitigates the effect by only trusting the `dim` field up to an amount we consider "acceptable".